server/tests/Infrastructure.UnitTests/Repositories/CourseRepositoryTests.cs

```diff
@@ -0,0 +1,154 @@
+using Microsoft.EntityFrameworkCore;
+using StudyZen.Application.Exceptions;
+using StudyZen.Domain.Entities;
+using StudyZen.Infrastructure.Persistence;
+
+namespace StudyZen.Infrastructure.UnitTests.Repositories;
+
+[TestFixture]
+public class CourseRepositoryTests
+{
+    private CourseRepository _courseRepository;
+    private ApplicationDbContext _dbContext;
+
+    [SetUp]
+    public void SetUp()
+    {
+        var options = new DbContextOptionsBuilder<ApplicationDbContext>()
+        .UseInMemoryDatabase(databaseName: Guid.NewGuid().ToString())
+        .Options;
+
+        _dbContext = new ApplicationDbContext(options);
+
+        _courseRepository = new CourseRepository(_dbContext);
+
+        AddTestData();
+    }
+
+    void AddTestData()
+    {
+        var course1 = new Course("Course1", "Description1");
```

Comment:
We could use `AddRange` method to add multiple instances; also, could call SaveChanges once